src/tcgwars/logic/impl/gen7/SunMoonPromos.groovy

```diff
@@ -2619,9 +2677,102 @@ public enum SunMoonPromos implements LogicCardInfo {
         return copy(UnbrokenBonds.LUCARIO_MELMETAL_GX_120, this);
       case GARCHOMP_GIRATINA_GX_SM193:
         return copy(UnifiedMinds.GARCHOMP_GIRATINA_GX_146, this);
+      case DETECTIVE_PIKACHU_SM194:
+        return basic (this, hp:HP090, type:L, retreatCost:2) {
+        weakness F
+        resistance M, MINUS20
+        move "Brilliant Deduction", {
+          text "Look at the top 4 cards of your deck and put 1 of them into your hand. Shuffle the other cards back into your deck."
+          energyCost L
+          attackRequirement {
+            assert my.deck : "Your deck is empty."
+          }
+          onAttack {
+            def top4 = my.deck.subList(0,4)
+            top4.showToMe("The top 4 cards of your deck.")
+            top4.select(max:1,"Choose a card to put in your hand.").moveTo(my.hand)
+            shuffleDeck()
+            }
```

Comment:
Can you check if this is an extra bracket?

Reply:
It is the ending bracket for the Brilliant Deduction's `onAttack` field.